### ISSUE TITLE
Store Upgrade fails with Legacy Indexes Created in 1.9 - Windows

### ISSUE THREAD

**Issue by gpierrick** [neo4j/neo4j#7042]
Hi,

Trying to upgrade a store from Neo4j 2.3.3 to Neo4j 3.0. I am guessing the exception is not due to the edition Community/Enterprise but I could be wrong.

I have tried @jexp suggestion to add dbms.record_format=standard but no luck

Windows 7
Java 8
neo4j 2.3.3 Community/Enterprise
Neo4j 3.0 Enterprise

See attached for exception.

How can I get those indexes migrated?

[legacy_indexes_exception_upgrade_exception_3_0.txt](https://github.com/neo4j/neo4j/files/245396/legacy_indexes_exception_upgrade_exception_3_0.txt)

**Comment by MishaDemianenko** [comment]
Hey @gpierrick, thx for reporting!

From what is visible in stack trace it's because of 
`Caused by: org.neo4j.upgrade.lucene.LegacyIndexMigrationException: Migration of legacy index at path:C:\Data\Projects\Ti
meblend\Neo4j\neo4j-community-3.0.0-RC1\data\databases\graph.db\upgrade\index\lucene\node\tags failed.`

It has nothing to do with `dbms.record_format` since it looks like to be related to lucene in some way or another.

Which version of lucene was used to build that legacy index? Does that folder contains valid index?

**Comment by gpierrick** [comment]
1.9.x was used

There are actually legacy indexes that I'd like to delete and 'tags' could be one of them

**Comment by MishaDemianenko** [comment]
@gpierrick  I was asking to make sure you use provided lucene version, so thx so far.
1.9 used same lucene version as 2.x so legacy index migration should work as expected. Can you please provide log files and failing index if that is possible? so we can help. 
Thx in advance.

**Comment by gpierrick** [comment]
Do you have an email address I can send the index that fails? - not sure which log file you would like to see?

**Comment by MishaDemianenko** [comment]
@gpierrick if you have already moved to 3.0 directory structure you should be able to find log files (neo4j.log and debug.log) in /neo4j_home/logs.
If you still end up with 2.3 folder layout - log files are: /neo4j_home/data/graph.db/messages.log and 
/neo4j_home/data/log/console.log.
It would be nice to have those.
Please send it to mikhaylo dot demianenko at neotechnology.com
Thx!

**Comment by gpierrick** [comment]
Sent!

**Comment by gpierrick** [comment]
Fixed! I deleted the index that it was failing on. @MishaDemianenko didn't see anything wrong with what I sent him apart from the index being in an invalid state.

**Comment by cybersam** [comment]
@gpierrick: How was this fixed? Other people are having this same problem. See http://stackoverflow.com/questions/37168718/neo4j-3-0-0-data-migration.

**Comment by gpierrick** [comment]
I ended up deleting the index that it was failing on. So it wasn't a fix as such

**Comment by MishaDemianenko** [comment]
@cybersam it was not possible to migrate one of the legacy index automatically. In that particular case index directory contained only one segment file, issue was resolved by removing that file. Hope this helps.

**Comment by tomb10516** [comment]
@MishaDemianenko I recommend that this issue be reopened until there is a fix for the bug or a workaround with step-by-step instructions is posted somewhere people will find it easily (without finding it on stack exchange and then following a link to a closed issue)

Some background:  I have never used any manual indexing and I think this is a bug whether it is in V2.3.3 or V3.0.1.  My database was created in V2.3.3

Here is what worked for me:
- The index file that was causing the failure was in the directory `~/neo4j-community-3.0.1/data-ca/graph.db/index/lucene/node/node_auto_index`   where `data-ca` is a copy of the data folder from my V2.3.3 installation.
- There were two files in that directory: segments_1  segments.gen
- I have no idea whether I was supposed to remove both or only one, but I removed both.
- I then followed the standard upgrade procedure and it worked.
